### DISCUSSION TITLE
Noob question: Do I need to add a CF worker to use CF proxy (orange cloud)?

### DISCUSSION THREAD

**Discussion by zenminimalist** [plausible/analytics#3456] [Self-Hosted Support]
Sorry if my question is a bit rudimentary. So speak to me as if I were a [golden retriever](https://www.youtube.com/watch?v=LTDS6SHwA6w) 😊

### Question
1. Is it possible to run PA behind a Cloudflare proxy (orange cloud turned **on** - and **not** DNS only or gray cloud)? Or does that defeat the purpose or mess with the tracking? If I open a private tab and go to my site via VPN from say Italy, I can see my visit from that country. 

I'm asking as I generally like protecting my sites that I have public and hide my server IP, but I'm confused if I need to add a worker mentioned here [Proxying Plausible through Cloudflare](https://plausible.io/docs/proxy/guides/cloudflare#:~:text=You%20can%20use%20Cloudflare%20Workers,is%20a%20free%20Cloudflare%20account.)

Or does that apply to the SaaS version of PA? 

**My setup**
I host my PA installation at `pa.domain0.com` and then track different domains such as `domain1.com, domain2.com`

**Comment by ruslandoga** [comment]
👋 @zenminimalist 

You can proxy through Cloudflare without using Workers. The IP address from `cf-connecting-ip` would [be used.](https://github.com/plausible/analytics/blob/2ada3d700fa694ed9c2eecd91f8bf16e9b86cb8f/lib/plausible_web/remote_ip.ex#L3) Workers are useful when you need to "rewrite" the paths where Plausible scripts are located, but they are not necessary.

**Reply by zenminimalist** [reply]
Ok, I don't have to change any setting on CF side as the the `cf-connecting-ip` is in the header and gets managed by CF. That correct?
